hax/treemakers/CorrectedDoubleS1Scatter.py

```diff
@@ -0,0 +1,386 @@
+            result['int_a_x_3d_' + algo] =\
+                result['int_a_x_' + algo] * (result['int_a_r_3d_' + algo] / result['int_a_r_' + algo])
+            result['int_a_y_3d_' + algo] =\
+                result['int_a_y_' + algo] * (result['int_a_r_3d_' + algo] / result['int_a_r_' + algo])
+
+            if abs(int_a_z) > abs(result['int_a_r_correction_3d_' + algo]):
+                result['int_a_z_3d_' + algo] = -np.sqrt(int_a_z ** 2 - result['int_a_r_correction_3d_' + algo] ** 2)
+            else:
+                result['int_a_z_3d_' + algo] = int_a_z
+
+            result['int_a_z_correction_3d_' + algo] = result['int_a_z_3d_' + algo] - int_a_z
+        
+        
+        # Int_b Position
+        
+        for algo in ['nn']:
+            result['int_b_r_correction_3d_' + algo] = self.fdc_3d_map.get_value(result['int_b_x_' + algo],
+                                                                          result['int_b_y_' + algo],
+                                                                          int_b_z)
+
+            result['int_b_r_3d_' + algo] = result['int_b_r_' + algo] + result['int_b_r_correction_3d_' + algo]
+            result['int_b_x_3d_' + algo] =\
+                result['int_b_x_' + algo] * (result['int_b_r_3d_' + algo] / result['int_b_r_' + algo])
+            result['int_b_y_3d_' + algo] =\
+                result['int_b_y_' + algo] * (result['int_b_r_3d_' + algo] / result['int_b_r_' + algo])
+
+            if abs(int_b_z) > abs(result['int_b_r_correction_3d_' + algo]):
+                result['int_b_z_3d_' + algo] = -np.sqrt(int_b_z ** 2 - result['int_b_r_correction_3d_' + algo] ** 2)
+            else:
+                result['int_b_z_3d_' + algo] = int_b_z
+
+            result['int_b_z_correction_3d_' + algo] = result['int_b_z_3d_' + algo] - int_b_z
+                 
+        # Apply LCE (light collection efficiency correction to s1)
+        
+        result['s1_int_a_xyz_correction_nn_fdc_3d'] = 1 / self.lce_map_nn_fdc_3d.get_value(result['int_a_x_3d_nn'],
+                                                                                     result['int_a_y_3d_nn'],
+                                                                                     result['int_a_z_3d_nn'])
+        
+        result['cs1_a_int_a'] = peaks[s1_a].area * result['s1_int_a_xyz_correction_nn_fdc_3d']
+        result['cs1_b_int_a'] = peaks[s1_b].area * result['s1_int_a_xyz_correction_nn_fdc_3d']
+
+        
+        result['s1_int_b_xyz_correction_nn_fdc_3d'] = 1 / self.lce_map_nn_fdc_3d.get_value(result['int_b_x_3d_nn'],
+                                                                                     result['int_b_y_3d_nn'],
+                                                                                     result['int_b_z_3d_nn'])
+        
+        result['cs1_b_int_b'] = peaks[s1_b].area * result['s1_int_b_xyz_correction_nn_fdc_3d']
```

Comment:
Missing cs1_a_int_b? 

I have to admit the naming confuses me but as long as the people using the tree can keep it straight that's fine.

Reply:
Yes, I change it by making the "int_a" correction the default one (cS1_a and cS1_b). 
Only "int_b" correction for S1_b signal is called with cs1_b_int_b. 
There is no cS1_a_int_b because it makes no sense to correct S1_a signal using int_b position.